crypto/x509/v3_utf8.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include "internal/cryptlib.h"
+#include <openssl/asn1.h>
+#include <openssl/conf.h>
+#include <openssl/x509v3.h>
+#include <crypto/x509v3.h>
+#include "ext_dat.h"
+
+const X509V3_EXT_METHOD v3_utf8_list[2] = {
+    EXT_UTF8STRING(NID_subjectSignTool),
+    EXT_UTF8STRING(NID_issuerSignTool)
+};
+
+char *i2s_ASN1_UTF8STRING(X509V3_EXT_METHOD *method,
+                          ASN1_UTF8STRING *utf8)
+{
+    char *tmp;
+
+    if (utf8 == NULL || utf8->length == 0) {
+        return NULL;
+    }
+    if ((tmp = OPENSSL_zalloc(utf8->length + 1)) == NULL) {
+        X509V3err(X509V3_F_I2S_ASN1_UTF8STRING, ERR_R_MALLOC_FAILURE);
+        return NULL;
+    }
+    memcpy(tmp, utf8->data, utf8->length);
+    return tmp;
+}
+
+ASN1_UTF8STRING *s2i_ASN1_UTF8STRING(X509V3_EXT_METHOD *method,
+                                     X509V3_CTX *ctx, const char *str)
+{
+    ASN1_UTF8STRING *utf8;
+    if (str == NULL) {
+        X509V3err(X509V3_F_S2I_ASN1_UTF8STRING,
+                  X509V3_R_INVALID_NULL_ARGUMENT);
+        return NULL;
+    }
+    if ((utf8 = ASN1_UTF8STRING_new()) == NULL) {
+        goto err;
```

Comment:
Seems strange to have a "goto err". But only one place where we actually goto it. I guess this is based on some pre-existing code....but still it looks odd. Probably I would just do the X509v3err call directly here and not have the goto.

Reply:
base on pre-existing code v3_ia5.c :)
fixed